tchannel-core/src/main/java/com/uber/tchannel/tracing/OpenTracingContext.java

```diff
@@ -85,6 +86,10 @@ public boolean hasSpan() {
     @Override
     @SuppressWarnings("resource")
     public void clear() {
+        if(scopeManager instanceof NoopScopeManager) {
```

Comment:
Why is this check needed? Please add comment.

Reply:
Code updated with a better check logic so that if other type of ScopeManager acts in the same way as NoopScopeManager, the logic can also avoid infinite loop.

Comment added.